src/data_source/storage/sql.rs

```diff
@@ -2219,8 +2219,10 @@ where
                 FROM header AS h
                 JOIN payload AS p ON
                     p.height = h.height
-                ORDER BY h.height DESC
-                LIMIT 50",
+                WHERE
+                    h.height IN (SELECT height FROM header ORDER BY height DESC LIMIT 50)
```

Comment:
It occurs to me that this will give the latest 50 blocks we have, even if they're non-consecutive (ie if we are missing a header for some reason). What will happen to the histogram in this case? Will there be a blank slot/any indication that data is missing? If not it could be misleading

Reply:
Indeed, no empty space would be inserted into the histogram, so it would show them as a contiguous list of blocks.  This may be potentially misleading, and could cause confusion upon inspection.

Reply:
Though, it does beg the question of how would we want to represent missing data in a block histogram?

Reply:
We could punt on the UI issue by just failing the endpoint if data is missing (e.g. you can do a quick scan over the `height` column of each row returned from the database and return an error if not consecutive). The browser should retry and the endpoint should eventually succeed once the missing data has been fetched.

This issue was already present though, so it can be considered out of scope for this PR, we can just create an issue to address it later